bmEditButtonPopup.lua

```diff
@@ -140,6 +144,8 @@ function BMButtonEditor:RenderEditButtonPopup()
                 BMSettings:GetSettings().Buttons[ButtonKey].Unassigned = nil                       -- clear the unassigned flag
 
                 BMSettings:SaveSettings(true)
+                BMSettings:updateButtonDB(btnUtils.shallowcopy(self.tmpButton), ButtonKey)

```

Comment:
Cant all the DB updates be handle from SaveSettings? How do you know when to call updatebutton/updateset/etc? That should ideally be abstracted away as much as possible.